dev/Repeater/FlexboxLayout.cpp

```diff
@@ -0,0 +1,468 @@
+        }
+
+        flexboxState = winrt::make_self<FlexboxLayoutState>();
+    }
+
+    context.LayoutStateCore(*flexboxState);
+}
+
+void FlexboxLayout::UninitializeForContextCore(winrt::LayoutContext const& context)
+{
+}
+
+winrt::Size FlexboxLayout::MeasureOverride(
+    winrt::NonVirtualizingLayoutContext const& context,
+    winrt::Size const& availableSize)
+{
+    auto state = context.LayoutState().as<FlexboxLayoutState>();
+    state->Rows.clear();
+
+    unsigned int itemsInRow = 0;
+    float growInRow = 0.0;
+    float shrinkInRow = 0.0;
+
+    float usedInCurrentMainAxis = 0;
+    float usedInCurrentCrossAxis = 0;
+
+    float usedMainAxis = 0;
+    float usedCrossAxis = 0;
+
+    auto completeRow = [&]()
+    {
+        FlexboxLayoutState::RowMeasureInfo newRow;
+        newRow.MainAxis = usedInCurrentMainAxis;
+        newRow.CrossAxis = usedInCurrentCrossAxis;
+        newRow.Count = itemsInRow;
+        newRow.Grow = growInRow;
+        newRow.Shrink = shrinkInRow;
+        state->Rows.emplace_back(newRow);
+
+        itemsInRow = 0;
+        growInRow = 0.0;
+        
+        shrinkInRow = 0.0;
+        usedMainAxis = std::max(usedMainAxis, usedInCurrentMainAxis);
+        usedInCurrentMainAxis = 0;
+        usedCrossAxis += usedInCurrentCrossAxis;
+        usedInCurrentCrossAxis = 0;
+
+
+
+    };
+   
+    
+
+    std::vector<winrt::UIElement> sortedChildren = ChildrenSortedByOrder(context.try_as<winrt::NonVirtualizingLayoutContext>());
+    for (winrt::UIElement const& child : sortedChildren)
+    {
+        
+
+        // Give each child the maximum available space
+        child.Measure(availableSize);
+        winrt::Size childDesiredSize = child.DesiredSize();
+
+
+        auto flexBasis = static_cast<float>((GetBasis(child)));
+        if (flexBasis > 0) {
+          
+            childDesiredSize = CreateSize(flexBasis, CrossAxis(childDesiredSize));
+        }
+
+
+            if (usedInCurrentMainAxis + MainAxis(childDesiredSize) > MainAxis(availableSize))
+            {
+                // Not enough space, time for a new row
+                if (IsWrapping() )
+                {
+                    completeRow();
+
+                    // It's possible that even making a new row won't work. Sorry, you're not going to fit!
+                    if (usedCrossAxis + CrossAxis(childDesiredSize) > CrossAxis(availableSize))
+                    {
+                        
+                        break;
+                    }
+                }
+                else
+                {
+ 
+                   break;
+                }
+            }
+     
+       
+        // Contribute our space
+        usedInCurrentMainAxis += MainAxis(childDesiredSize);
+        usedInCurrentCrossAxis = std::max(usedInCurrentCrossAxis, CrossAxis(childDesiredSize));
+        itemsInRow++;
+        growInRow += float(GetGrow(child));
+
```

Comment:
cast goes away when you make these doubles

Reply:
I keep getting a compiler error that states that "potential loss of data converting double to float" 

This is why I originally casted the value as a float before any computations.